node/src/components/small_network.rs

```diff
@@ -953,6 +884,67 @@ where
                     Effects::new()
                 }
             }
+            Event::ContractRuntimeAnnouncement(
+                ContractRuntimeAnnouncement::LinearChainBlock(_)
+                | ContractRuntimeAnnouncement::StepSuccess { .. },
+            ) => Effects::new(),
+            Event::ContractRuntimeAnnouncement(
+                ContractRuntimeAnnouncement::UpcomingEraValidators {
+                    era_that_is_ending,
+                    mut upcoming_era_validators,
+                },
+            ) => {
+                if era_that_is_ending < self.active_era {
+                    debug!("ignoring past era end announcement");
+                } else if era_that_is_ending != self.active_era {
```

Comment:
```suggestion
                } else if era_that_is_ending > self.active_era {
```

Reply:
The if else if else branch here looks like (or should look like)

```
IF ending era is past era
   ignore
ELSE IF ending era is current era
   do regular update
ELSE (implies "ending era is future era")
   clear all limiters, as we cannot rely on the ending era being accurate
   and only use the information in we just received
```

The proposed change at the very least makes the `else` block at the dead code, if I am not mistaken (since one of `era_that_is_ending < self.active_era` and `era_that_is_ending >= self.active_era` should always be true).

Reply:
This is now obsolete, as the code has been rewritten in c5720c575